kasa/iot/modules/module.py

```diff
@@ -34,6 +35,14 @@ class IotModule(ABC):
     def __init__(self, device: "IotDevice", module: str):
         self._device = device
         self._module = module
+        self._module_descriptors: Dict[str, Descriptor] = {}
+
+    def add_descriptor(self, desc):
```

Comment:
Rather than module adding Descriptors/Features can't Module inherit from Feature.  That way we'd just need to implement the Feature attributes in the base module and derived modules and consumers will call them like features.  Or composition where an iot implementation of 'Feature' takes a module as a parameter and calls the methods on the Module?

Reply:
I don't think we should inherit directly from a module, as a module can implement a variety of things. Consider the auto-off feature: it consists of a switch (on/off), a setting (in how many minutes the device should be turned off), and a sensor ("device will be turned off at <timestamp>". For that reason, it feels at least currently more natural to me, that a module holds the features and not the other way around.